service/common/src/main/java/org/apache/polaris/service/ratelimiter/RateLimiterFilter.java

```diff
@@ -29,6 +33,9 @@
 
 /** Request filter that returns a 429 Too Many Requests if the rate limiter says so */
 @Provider
+@PreMatching
+@Priority(Priorities.USER)
```

Comment:
The original Dropwizard filter did not specify any priority. Accordding to JAX-RS, this means priority `USER`. I think it's better to be clear about this and specify the priority.

Reply:
Using `USER` means that this filter cannot be abused by unauthenticated requests – but also means, that malicious requests will hit the authenticator filter first, thus defeating the purpose of a rate limiter.

Reply:
Good point. Still, if the authenticator is merely validating access tokens, that should be fine, I guess... With JWT, the validation effort is not much higher than the effort to keep track of the request rate, I guess.

That said, I think it would be worth moving the loading of principal information out the the authentication code and do that after all request checks are performed (including the rate limit filter).

Reply:
You make good points as well. We could also consider this Quarkus extension:

https://docs.quarkiverse.io/quarkus-bucket4j/dev/index.html

Although, it's not a filter, but rather applies at REST resource level.